### PR TITLE
Support portmapping

### PR DESCRIPTION
**Pull request by heartlock**
What this PR does:
- [x] support port mapping
- [x] enable e2e test for port mapping
- [x] update README

@feiskyer Thanks for your help! Ready for review.
Closes: #122

### CODE REVIEW

File: pkg/hyper/network.go [hunk 1]
```diff
@@ -23,8 +23,19 @@ import (
...
+	"k8s.io/kubernetes/pkg/kubelet/network/hostport"
 )
 
+// cniPortMapping maps to the standard CNI portmapping Capability
+// see: https://github.com/containernetworking/cni/blob/master/CONVENTIONS.md
+type cniPortMapping struct {
+	HostPort      int32  `json:"hostPort"`
+	ContainerPort int32  `json:"containerPort"`
+	Protocol      string `json:"protocol"`
+	HostIP        string `json:"hostIP"`
```
Comment (feiskyer) [inline pkg/hyper/network.go]: `hostIP,omitempty`

File: hack/test-e2e.sh [hunk 2]
```diff
@@ -88,32 +88,44 @@ function install_remote_hyperd() {
...
+                "subnet": "10.30.0.0/16",
+                "routes": [
+                    { "dst": "0.0.0.0/0"   }
+                ]
+            }
+        },
+        {
+            "type": "portmap",
+            "capabilities": {"portMappings": true},
+            "snat": false
```
Comment (feiskyer) [inline hack/test-e2e.sh]: should be true here, refer https://github.com/containernetworking/plugins/tree/master/plugins/meta/portmap#snat

File: hack/test-e2e.sh [hunk 3]
```diff
@@ -88,32 +88,44 @@ function install_remote_hyperd() {
 function configure_cni() {
   # install cni
```
Comment (feiskyer) [inline hack/test-e2e.sh]: could you also update the CNI setup in README.md?

Comment (heartlock) [inline hack/test-e2e.sh]: will do

**Pull request merged**